### ISSUE TITLE
Error while using userless authentication

### ISSUE THREAD

**Issue by garretyoder** [mattbdean/JRAW#245]
It seems that JRAW is not automatically storing a refresh token while using userless authentication. 

`java.lang.IllegalStateException: This client is logged out and should not be used anymore`

```kotlin
Reddit.accountHelper.switchToUserless()
System.out.println(Reddit.client.user("user").about().commentKarma)
```

Will yield the above error, however forcing a complete clear of the tokenstore will allow it to pull a new token and authenticate, until the next launch.

```kotlin
Reddit.tokenStore.clear()
Reddit.accountHelper.switchToUserless()
System.out.println(Reddit.client.user("user").about().commentKarma)
```

JRAW Setup:
```kotlin
class Reddit : Application() {

    companion object {
        lateinit var accountHelper: AccountHelper
            private set
        lateinit var tokenStore: SharedPreferencesTokenStore
            private set
        val client
            get() = accountHelper.reddit
    }



    override fun onCreate() {
        super.onCreate()
        val provider: AppInfoProvider = ManifestAppInfoProvider(applicationContext)
        val deviceUUID: UUID = UUID.randomUUID()

        tokenStore = SharedPreferencesTokenStore(applicationContext)
        tokenStore.autoPersist=true
        tokenStore.load()


        accountHelper = AndroidHelper.accountHelper(provider, deviceUUID, tokenStore)
        
        accountHelper.onSwitch { it.logger = SimpleHttpLogger(SimpleHttpLogger.DEFAULT_LINE_LENGTH, SimpleAndroidLogAdapter(Log.INFO)) }
    }

}
```

**Comment by garretyoder** [comment]
This issue is also reproducable in the JRAW example app by doing nothing but making the same call noted above.

**Comment by garretyoder** [comment]
So I have solved the issue. The issue seems to have actually been fixed somewhere inbetween JRAW 1.0.0 and the current source set. The issue is present in 1.0.0, but is resolved in the current source. I found this while compiling the project locally to try to debug. I'm not sure which commit or issue represented this problem being fixed, but I highly recommend pushing a new release to both JRAW and JRAW-Android that way people can use the fixed version without the workaround I will detail below.

The issue was that even if I pulled JRAW-Android snapshot off jitpack, it was still linking to 1.0.0. So for anyone else encountering this problem the workaround I'm using is to compile JRAW-Android excluding JRAW, then loop in the most recent JRAW source.

```    //JRAW
    implementation ('com.github.mattbdean:JRAW-Android:-SNAPSHOT'){
        exclude group: 'net.dean.jraw'
    }
    implementation 'com.github.mattbdean:JRAW:-SNAPSHOT'
```
